qemu/tests/cpu_info_check.py

```diff
@@ -0,0 +1,129 @@
+        # TODO: Need be updated when there is new cpu model introduced
+    if host_qemu in VersionInterval('[,4.2.0)'):
+        remove_models(params.objects('cpu_model_8'))
+    if host_qemu in VersionInterval('[,3.1.0)'):
+        remove_models(params.objects('cpu_model_3_1_0'))
+    if host_qemu in VersionInterval('[,2.12.0)'):
+        remove_models(params.objects('cpu_model_2_12_0'))
+    qemu_binary = utils_misc.get_qemu_binary(params)
+    logging.info('Query cpu models by qemu command')
+    query_cmd = "%s -cpu ? | awk '{print $2}'" % qemu_binary
+    qemu_binary_output = process.system_output(
+        query_cmd, shell=True).decode().splitlines()
+    cpuid_index = qemu_binary_output.index('CPUID')
+    cpu_models_binary = qemu_binary_output[1: cpuid_index - 1]
+    cpu_flags_binary = qemu_binary_output[cpuid_index + 1:]
+    params['start_vm'] = 'yes'
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+    vm = env.get_vm(vm_name)
+    # query cpu model supported by qemu
+    logging.info('Query cpu model supported by qemu by qemu monitor')
+    qmp_model_output = str(vm.monitor.cmd('qom-list-types'))
+    qmp_def_output = str(vm.monitor.cmd('query-cpu-definitions'))
+
+    # Check if all the output contain expected cpu models
+    output_list = {'qemu-kvm': cpu_models_binary,
+                   'qom-list-types': qmp_model_output,
+                   'query-cpu-definitions':  qmp_def_output}
+    missing = dict.fromkeys(output_list.keys(), [])
+    for cpu_model in cpu_types:
+        logging.info('Check cpu model %s from qemu command output and'
+                     ' qemu monitor output' % cpu_model)
+        for key, value in output_list.items():
+            if cpu_model not in value:
+                missing[key].append(cpu_model)
+    for key, value in missing.items():
+        if value:
+            test.fail('%s is missing in the %s output: %s\n' %
+                      (', '.join(value), key, output_list[key]))
+
+    # Check if qemu command output matches qmp output
+    missing = []
+    logging.info('Check if qemu command output matches qemu monitor output')
+    for cpu_model in cpu_models_binary:
+        if cpu_model not in qmp_model_output:
+            missing.append(cpu_model)
+    if missing:
+        test.fail('The qemu monitor output does not included all the cpu'
+                  ' model in qemu command output, missing: \n %s' %
+                  ', '.join(missing))
+
+    # Check if the flags in qmp output matches expectation
+    args = {'type': 'full', 'model': {'name': vm.cpuinfo.model}}
+    output = vm.monitor.cmd('query-cpu-model-expansion', args)
+    model = output.get('model')
+    model_name = model.get('name')
+    if model_name != vm.cpuinfo.model:
+        test.fail('Command query-cpu-model-expansion return'
+                  ' wrong model: %s' % model_name)
+    model_prop = model.get('props')
```

Comment:
Hi， the flag name that we add to CPU may be different from the flag which shows on the host like that host shows flag `avx512_vnni` but our query shows flag `avx512vnni `, so we can just check the `model_name` and don't to check the `mode_prop`.

Reply:
Hi @nanliu-r Updated, please help check, thanks.